src/modules/privacy-idea-administration/privacy-idea-administration.controller.ts

```diff
@@ -89,4 +92,57 @@ export class PrivacyIdeaAdministrationController {
         );
         return new TokenStateResponse(piToken);
     }
+
+    @Post('assign/hardwareToken')
+    @HttpCode(HttpStatus.OK)
+    @ApiCreatedResponse({
+        description: 'The hardware token was successfully assigned.',
+        type: AssignHardwareTokenResponse,
+    })
+    @ApiBadRequestResponse({ description: 'Not found.' })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to assign hardware token.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to reset token.' })
+    @ApiNotFoundResponse({ description: 'Insufficient permissions to assign hardware token.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while assigning a hardware token.' })
+    @Public()
+    public async assignHardwareToken(
+        @Body() params: AssignHardwareTokenBodyParams,
+    ): Promise<AssignHardwareTokenResponse | undefined> {
+        try {
+            const result: AssignTokenResponse = await this.privacyIdeaAdministrationService.assignHardwareToken(
+                params.serial,
+                params.otp,
+                params.user,
+            );
+            return new AssignHardwareTokenResponse(
+                result.id,
+                result.jsonrpc,
+                result.time,
+                result.version,
+                result.versionnumber,
+                result.signature,
+                'Token wurde erfolgreich zugeordnet.',
+            );
+        } catch (error) {
+            if (error instanceof TokenError) {
+                // Return structured error response with code and message
+                throw new HttpException(
+                    {
+                        statusCode: HttpStatus.BAD_REQUEST,
+                        message: error.message,
+                        code: error.name,
+                    },
+                    HttpStatus.BAD_REQUEST,
+                );
+            }
+            // Handle other unexpected errors
+            throw new HttpException(
+                {
+                    statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+                    message: 'An unexpected error occurred.',
+                },
+                HttpStatus.INTERNAL_SERVER_ERROR,
+            );
```

Comment:
We try to use Controller-Specific error-filters so we don't have to do things like this. 
i.e. something like `PersonExceptionFilter`. Maybe add an ExceptionFilter to the entire PrivacyIdeaAdministrationController to handle errors that can occur in here.

Reply:
PrivacyIdeaAdministrationExceptionFilter implemented -> Frontend-Part work in progress